packages/devtools_app/lib/src/screens/profiler/cpu_profile_controller.dart

```diff
@@ -207,15 +207,19 @@ class CpuProfilerController
     if (analyticsScreenId != null) {
       // Pull and process cpu profile data [pullAndProcessHelper] and time the
       // operation for analytics.
-      await ga.timeAsync(
-        analyticsScreenId!,
-        analytics_constants.cpuProfileProcessingTime,
-        asyncOperation: pullAndProcessHelper,
-        screenMetricsProvider: () => ProfilerScreenMetrics(
-          cpuSampleCount: cpuProfiles.profileMetaData.sampleCount,
-          cpuStackDepth: cpuProfiles.profileMetaData.stackDepth,
-        ),
-      );
+      try {
+        await ga.timeAsync(
+          analyticsScreenId!,
+          analytics_constants.cpuProfileProcessingTime,
+          asyncOperation: pullAndProcessHelper,
+          screenMetricsProvider: () => ProfilerScreenMetrics(
+            cpuSampleCount: cpuProfiles.profileMetaData.sampleCount,
+            cpuStackDepth: cpuProfiles.profileMetaData.stackDepth,
+          ),
+        );
+      } on ProcessCancelledException catch (_) {
```

Comment:
we are now handling ProcessCancelledException in both the try and the if block. Condense so that we are only handling it once. (wrap the try around the if/else)